### PR TITLE
Attempt CodeCov integration

### PR DESCRIPTION
**Pull request by davidjgonzalez**

### CODE REVIEW

**Comment by davidjgonzalez** [comment]
@kwin I think the CodeCov works on this (removed coveralls) - can you check the SNAPSHOT to OSSRH piece (which should only trigger on master)

**Comment by davidjgonzalez** [comment]
I assume the CodeCov complaining about "no report found to compare against" gets resolved when this gets into master and branches are made off that.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -34,30 +34,51 @@ jobs:
...
+      - name: Upload code coverage to CodeCov (Main build)
+        if: ${{ matrix.isMainBuildEnv }}
+        run: bash <(curl -s https://codecov.io/bash)
+        
+      - name: Deploy SNAPSHOT to OSSRH     
+        if: ${{ matrix.isMainBuildEnv && github.ref == 'refs/heads/main' }}
+        env:
+          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
+        run: mvn clean deploy
```
Comment (kwin) [inline .github/workflows/maven.yml:84]: `mvn clean deploy` encompasses a regular Maven build.

**Comment by kwin** [comment]
I think we should keep the approach of handling differences between branches in environment variables. Just add the coveralls upload step conditionally in addition to the Maven build leveraging env variables

File: .github/workflows/maven.yml (hunk 1 above)
Comment (davidjgonzalez) [reply .github/workflows/maven.yml:84]: you thinking just do a `mvn deploy:deploy --batch-mode` ?

Comment (kwin) [reply .github/workflows/maven.yml:84]: No, just leave as before this PR and do the coveralls upload post deployment.

Comment (davidjgonzalez) [reply .github/workflows/maven.yml:84]: Your master still have coveralls in it right? So that needs to come out

**Pull request closed without merging**